### PR TITLE
* DDO-1468 Get apps to use ingress library for Ingress

### PR DESCRIPTION
**Pull request by knapii-developments**
Converts apps to use Ingress library. 

Checks for diffs in prod(**Notice how arrangement is changed**):
[Agora](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3445584626#step:8:709)
[Buffer](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3445584626#step:8:804)
[Consent](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3462772563#step:8:1004)
[Cromiam](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3445584626#step:8:1151)
[Firecloudoch](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3445584626#step:8:1324)
[Firecloudui](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3445584626#step:8:1483)
[Jobmanager](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3445584626#step:8:1601)
[Rawls](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3462651665#step:8:2063)
[Leonardo](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3445584626#step:8:1806)
[Thurloe](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3445584626#step:8:2040)
[Workspacemanager](https://github.com/broadinstitute/terra-helmfile/pull/1583/checks?check_run_id=3445584626#step:8:2228)

The PR satisfies DDO-1468 when released with [pull #1583 in terra-helmfile](https://github.com/broadinstitute/terra-helmfile/pull/1583).

### CODE REVIEW

**Review by jyang-broad — COMMENTED** [review]
I like the changes, much cleaner! I think there are a couple breaking references though.

File: charts/workspacemanager/Chart.yaml [hunk 1]
```diff
@@ -16,3 +16,7 @@ dependencies:
     condition: postgres.enabled
     version: 0.1.0
     repository: https://broadinstitute.jfrog.io/broadinstitute/terra-helm-proxy
+  - name: ingresslib
+    condition: ingress.enabled
+    version: 0.3.0
+    repository: file://../ingresslib
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/workspacemanager/Chart.yaml:22]: Yes it should. 4ed2e346

File: charts/firecloudui/templates/ingress/ingress.yaml [hunk 2]
```diff
@@ -1,23 +1,3 @@
...
-{{ include "firecloudui.labels" . | nindent 4 }}
-  annotations:
-    networking.gke.io/v1beta1.FrontendConfig: "{{ .Values.name }}-ingress-frontendconfig"
-    kubernetes.io/ingress.global-static-ip-name: {{ required "ingress.staticIpName value is required" .Values.ingress.staticIpName | quote }}
-    kubernetes.io/ingress.allow-http: "false"
-{{- if not (empty .Values.ingress.cert.preSharedCerts) }}
-    ingress.gcp.kubernetes.io/pre-shared-cert: {{ .Values.ingress.cert.preSharedCerts | join "," | quote }}
-{{- end }}
-spec:
-{{- if empty .Values.ingress.cert.preSharedCerts }}
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/firecloudui/templates/ingress/ingress.yaml:16]: Some applications have their own tls certificate logic. When they do, I leave them in the app chart so their logic/values persist. In this instance, I didn't carry the logic over. 4ed2e346

File: ingresslibchart/.helmignore [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply ingresslibchart/.helmignore:1]: Yes it is!

File: charts/firecloudui/templates/ingress/ingress.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/firecloudui/templates/ingress/ingress.yaml:16]: Here's [another example](https://github.com/broadinstitute/terra-helm/blob/3c514581101bcc232861ebeffe6ed5cd38967229/charts/rawls/templates/ingress.yaml#L4) where a value is defined differently than in the library chart. The app chart can override or add additional values in the library.

File: charts/firecloudorch/templates/ingress/ingress.yaml [hunk 4]
```diff
@@ -1,23 +1,8 @@
...
-    networking.gke.io/v1beta1.FrontendConfig: "{{ .Values.name }}-ingress-frontendconfig"
-    kubernetes.io/ingress.global-static-ip-name: {{ required "ingress.staticIpName value is required" .Values.ingress.staticIpName | quote }}
-    kubernetes.io/ingress.allow-http: "false"
-{{- if not (empty .Values.ingress.cert.preSharedCerts) }}
-    ingress.gcp.kubernetes.io/pre-shared-cert: {{ .Values.ingress.cert.preSharedCerts | join "," | quote }}
-{{- end }}
+{{- include "libchart.ingress" (list . "firecloudorch.ingress" )  -}}
+{{- define "firecloudorch.ingress" -}}
 spec:
 {{- if empty .Values.ingress.cert.preSharedCerts }}
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/firecloudorch/templates/ingress/ingress.yaml:4]: See [comment](https://github.com/broadinstitute/terra-helm/pull/457#discussion_r699326221)

File: charts/ingresslibchart/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v2
+name: ingress-library
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/ingresslibchart/Chart.yaml]: This was leftover POC work. Thanks for catching this!

File: charts/firecloudui/templates/ingress/ingress.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jyang-broad) [reply charts/firecloudui/templates/ingress/ingress.yaml:16]: Oooh got it.

**Review by jyang-broad — APPROVED** [review]

**Pull request merged**